### PR TITLE
Added optional filebeat extension to eb_deploy script

### PR DESCRIPTION
**Pull request by carsongee**

### CODE REVIEW

File: shell/filebeat-eb-extension.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+done
+
+cat << EOF > "$FB_EXTENSION_PATH"
+files:
+    "/etc/filebeat/filebeat.yml":
+        mode: "0644"
+        owner: root
+        group: root
+        content: |
+          filebeat:
```
Comment (carsongee) [inline shell/filebeat-eb-extension.sh]: :yodawg: I heard you like yaml, so I put a yaml file in your bash file with an embedded yaml file in it.

File: shell/filebeat-eb-extension.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# LOGSTASH_DOC_TYPE: Log type tagging from filebeat to logstash
+# FB_EXTENSION_PATH: Path to filebeat extension configuration to generate
+
+LOGSTASH_DOC_TYPE=${LOGSTASH_DOC_TYPE:-elasticbeanstalk}
+FB_EXTENSION_PATH=${FB_EXTENSION_PATH:-.ebextensions/100-filebeat.config}
+
+indent_cert() {
+    # shellcheck disable=SC2001
+    echo "$1" | sed 's/^/        /'
+}
```
Comment (gossnj) [inline shell/filebeat-eb-extension.sh]: Cute 🐶

File: shell/filebeat-eb-extension.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      content: |
+$(indent_cert "$LOGSTASH_CLIENT_KEY")
+
+commands:
+  100_command:
+    command: "curl -L -O https://download.elastic.co/beats/filebeat/filebeat-1.2.3-x86_64.rpm"
+    test: "[ ! -f filebeat-1.2.3-x86_64.rpm ]"
+    cwd: "/home/ec2-user"
+  200_command:
+    command: "rpm -ivh --replacepkgs filebeat-1.2.3-x86_64.rpm && touch /tmp/installed-filebeats"
```
Comment (gossnj) [inline shell/filebeat-eb-extension.sh]: Kinda sucks to have these versions hardcoded into the script, but I guess won't be upgrading them that frequently?

**Comment by gossnj** [comment]
Generally it seems fine. This was my experience with .ebextensions, which is that they're ugly

File: shell/filebeat-eb-extension.sh (hunk 3 above)
Comment (carsongee) [inline shell/filebeat-eb-extension.sh]: Good idea, I can make it a default that is overridable with a var especially since 5.0 will be out very soon

**Comment by carsongee** [comment]
Added the version configuration

**Pull request merged**